### PR TITLE
Development build documentation

### PR DESCRIPTION
**Pull request by jmaas-onoffice**
Document how to do a development build and improve the scripts.

### CODE REVIEW

**Comment by jmaas-onoffice** [comment]
Let us test that the changes to the build script work by giving an approving review. ;)

File: .github/workflows/dev-build.yml [hunk 1]
```diff
@@ -35,24 +35,19 @@ jobs:
       run: composer install
 
     - name: Build for development
-      run: PREFIX=/tmp/release make test-zip
```
Comment (jayay) [inline .github/workflows/dev-build.yml:38]: `make release` names the plugin _onOffice for WP-Websites_ which can lead to the issue, that a WP instance that has the pre-release version of the plugin installed, can update itself and overwrite it with the version from wordpress.org. This is the reason why there is the `make test-zip` target: it holds on the the plugin name _onOffice for WP-Websites *(dev)*_. `make test-zip` definitely needs to stay.

File: .github/workflows/dev-build.yml [hunk 2]
```diff
@@ -35,24 +35,19 @@ jobs:
...
-    - run: cat /tmp/release/onoffice/plugin.php
-
     - name: Zip files
       run: (cd /tmp/release/ && zip -r ./onoffice-${{ env.DEV_BUILD_VERSION}}.zip ./onoffice/)  # To prevent nested paths (like /tmp/release/onoffice/readme.txt) we have to cd into the directory.
 
     - name: Upload zip
       uses: actions/upload-artifact@v3
       with:
         name: onoffice-${{ env.DEV_BUILD_VERSION }}-please-unpack
-        path: /tmp/release/onoffice-${{ env.DEV_BUILD_VERSION }}.zip  # Uploading a .zip leads to a .zip inside a .zip, but uploading the folder takes significantly longer.
```
Comment (jayay) [inline .github/workflows/dev-build.yml:55]: the old version was fine

**Review by jayay — CHANGES_REQUESTED** [review]

File: BUILDING.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Building
+
+This is how you can generate a .zip to upload to a WordPress instance.
+
+1. Run `composer install`.
+    - If it asks you to update, run `composer update`.
```
[the remark this answers is not in the document]
Comment (jmaas-onoffice) [reply BUILDING.md:6]: Thanks for the explanation!

File: .github/workflows/dev-build.yml (hunk 1 above)
Comment (jmaas-onoffice) [reply .github/workflows/dev-build.yml:38]: Oh, I overlooked that release changes the name. Then we'll simply .zip it twice.

File: .github/workflows/dev-build.yml (hunk 2 above)
Comment (jmaas-onoffice) [reply .github/workflows/dev-build.yml:55]: Do you mean the comment? I'm not sure it is self-explanatory, so I tried to provied a bit more context.

File: BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Building
+
+This is how you can generate a .zip to upload to a WordPress instance.
+
+1. Run `composer install`.
```
Comment (jayay) [inline BUILDING.md:5]: I would prefer something like
__You'll need the PHP extensions `mbstring`, `curl` and `simplexml`. If you're using Ubuntu, you can install these by running `sudo apt install php-mbstring php-curl php-simplexml`__ just to not pretend Ubuntu is the only operating system out there.

Comment (jayay) [reply BUILDING.md:5]: Also, running `apt install` for a package that's already installed doesn't hurt, so there is not really a reason not to have them all in one line

**Review by jayay — COMMENTED** [review]
I commented on some things. Nothing mandatory, just thoughts.

File: BUILDING.md (hunk 4 above)
Comment (jmaas-onoffice) [reply BUILDING.md:5]: How can we check which extensions are required?

**Review by onOffice-Web — APPROVED** [review]
Test approval for action.

**Review by onOffice-Web — APPROVED** [review]
Test approval for action.

**Review by onOffice-Web — APPROVED** [review]
Test approval for action.

**Review by onOffice-Web — APPROVED** [review]
Test approval for action.

**Pull request merged**